ironfish/src/workerPool/pool.ts

```diff
@@ -316,15 +316,15 @@ export class WorkerPool {
  * Calculates the number of workers to use based on machine's number of cpus
  */
 export function calculateWorkers(nodeWorkers: number, nodeWorkersMax: number): number {
-  let workers = nodeWorkers
-  if (workers === -1) {
-    workers = os.cpus().length - 1
+  // The -1 in the calculation for both `workers` and `maxWorkers` is to allow
+  // room to the node main process, as well as to reduce the impact on the
+  // user's system responsiveness
 
-    const maxWorkers = nodeWorkersMax
-    if (maxWorkers !== -1) {
-      workers = Math.min(workers, maxWorkers)
-    }
-  }
+  // TODO: ideally we should use `os.availableParallelism()` here, instead of
+  // `os.cpus().length`, but it seems like Typescript doesn't have the type
+  // bindings for it
```

Comment:
I thought this too, but I think it was only added in 18.14.0: https://nodejs.org/api/os.html#osavailableparallelism and we technically support all versions of 18. You could check if it exists at runtime and call it, else fall back to os.cpus().length

Reply:
Somehow I was convinced we jumped to Node 20, but no, you're right: we're still on Node 18